mhs-reference-implementation/mhs/routing/sds.py

```diff
@@ -56,62 +44,70 @@ def __init__(self, ldap_address: str = None, tls: bool = False):
 
         details = await self._mhs_details_lookup(party_key, interaction_id)
 
         if len(details) > 1:
-            logging.warning(f"More than one mhs details  returned on inputs: "
-                            f"ods: {ods_code} - interaction: {interaction_id}")
+            logger.warning(f"More than one mhs details returned on inputs: "
+                           f"ods: {ods_code} - interaction: {interaction_id}")
         return details[0]['attributes']
 
-    async def _accredited_system_lookup(self, ods_code: str, interaction_id: str):
+    async def _accredited_system_lookup(self, ods_code: str, interaction_id: str) -> List:
         """
         Used to find an accredited system, the result contains the nhsMhsPartyKey.
         This can then be used to find an MHS endpoint
         :return: endpoint details - filtered to only contain nhsMHSPartyKey
         """
-        if not self.connection:
-            raise IOError('No connection established')
 
         search_filter = f"(&(nhsIDCode={ods_code}) (objectClass={AS_OBJECT_CLASS}) (nhsAsSvcIA={interaction_id}))"
 
         message_id = self.connection.search(search_base=NHS_SERVICES_BASE,
                                             search_filter=search_filter,
-                                            attributes="nhsMHSPartyKey")
+                                            attributes=MHS_PARTY_KEY)
+        logger.info(f'Message id - {message_id} - for query: ods code - {ods_code} '
+                    f': interaction id - {interaction_id}')
 
         query_task = self._get_query_result(message_id)
         response = await query_task
         return response
 
-    async def _mhs_details_lookup(self, party_key: str, interaction_id: str):
+    async def _mhs_details_lookup(self, party_key: str, interaction_id: str) -> List:
         """
         Given a party key and a service, this will return an object containing the attributes of that party key,
         including the endpoint address
         :param party_key:
         :param interaction_id:
         :return: all the endpoint details
         """
-        if not self.connection:
-            raise IOError('No connection established')
-
-        search_filter = f"(&(objectClass={MHS_OBJECT_CLASS}) (nhsMhsPartyKey={party_key}) (nhsMhsSvcIA={interaction_id}))"
+        search_filter = f"(&(objectClass={MHS_OBJECT_CLASS})" \
+            f" ({MHS_PARTY_KEY}={party_key})" \
+            f" (nhsMhsSvcIA={interaction_id}))"
         message_id = self.connection.search(search_base=NHS_SERVICES_BASE,
                                             search_filter=search_filter,
                                             attributes=mhs_attributes)
 
+        logger.info(f'Message id - {message_id} - for query: party key - {party_key} '
+                    f': interaction id - {interaction_id}')
+
         query_task = self._get_query_result(message_id)
         response = await query_task
```

Comment:
I think it's worth having a log message to identify that the query was successful here. This would apply to the `_accredited_system_lookup` method too.

Reply:
Added this